.ci/package-storage-publish.groovy

```diff
@@ -31,7 +45,30 @@ pipeline {
         pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
         deleteDir()
         gitCheckout(basedir: "${BASE_DIR}")
-        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+        stash(allowEmpty: true, name: 'source', useDefaultExcludes: false)
+      }
+    }
+    stage('Build package') {
+      steps {
+        cleanup()
+        withGoEnv() {
+          dir("${BASE_DIR}") {
+            sh(label: 'Install elastic-package',script: "make install")
+            // sh(label: 'Install elastic-package', script: 'go build github.com/elastic/elastic-package')
+            dir("test/packages/package-storage/package_storage_candidate") {
+              sh(label: 'Build package', script: "elastic-package build -v --zip")
+            }
+          }
+        }
+        stash(allowEmpty: true, name: 'build-package', includes: "${BASE_DIR}/build/integrations/*.zip", useDefaultExcludes: false)
+      }
+    }
+    stage('Sign and publish package') {
+      steps {
+        cleanup(source: 'build-package')
+        dir("${BASE_DIR}") {
+          packageStoragePublish('build/integrations')
```

Comment:
I'd vote for moving this to a shared library, so signing and publishing in any repository is basically a one line :)
We would need to see how to pass all the parameters in case environment is not recommended.

Reply:
I guess that it would be a big input map with all properties :) I'm ok with moving it there, not sure if I should do this now or later. WDYT?

I guess it's a matter of priorities: indexing & package registry implementation vs. support in apm-pipeline-library

Reply:
Yes, do it later, no need to do it now.